root.c

```diff
@@ -2311,8 +2311,18 @@ static w_root_t *root_resolve(const char *filename, bool auto_watch,
   *created = true;
```

Comment:
I think it'd be clearer to move this down inside the `pthread_mutex_lock(&root_lock)` and setting it only in the `else` block. (It's not like a pointer dereference under the lock is going to cost much at all compared to the `w_ht_get()` and `w_ht_set()` calls..)